File: lib/data/species_catalog.dart
class BirdSpecies {
  final String commonName;
  final String scientificName;
  final String family;
  final String statusIUCN;
  final bool isEndemic;
  final bool isMigratorial;

  const BirdSpecies({
    required this.commonName,
    required this.scientificName,
    required this.family,
    required this.statusIUCN,
    this.isEndemic = false,
    this.isMigratorial = false,
  });
}

class SpeciesCatalog {
  static const families = [
    "Trochilidae (Colibríes)",
    "Ramphastidae (Tucanes)",
    "Psittacidae (Loros y Guacamayas)",
    "Cotingidae (Gallitos y Cotingas)",
    "Falconidae (Halcones y Caracaras)",
    "Columbidae (Palomas y Tórtolas)",
    "Thraupidae (Tangaras y semilleros)",
    "Turdidae (Mirlas y tordos)",
    "Ardeidae (Garzas)",
    "Cathartidae (Buitres y Cóndores)"
  ];

  static const species = [
    // Trochilidae
    BirdSpecies(
        commonName: "Colibrí rutilante",
        scientificName: "Colibri coruscans",
        family: "Trochilidae (Colibríes)",
        statusIUCN: "Preocupación Menor (LC)"),
    BirdSpecies(
        commonName: "Colibrí picoespada",
        scientificName: "Ensifera ensifera",
        family: "Trochilidae (Colibríes)",
        statusIUCN: "Preocupación Menor (LC)",
        isEndemic: true),
    BirdSpecies(
        commonName: "Colibrí coliancho",
        scientificName: "Selasphorus platycercus",
        family: "Trochilidae (Colibríes)",
        statusIUCN: "Preocupación Menor (LC)",
        isMigratorial: true),

    // Ramphastidae
    BirdSpecies(
        commonName: "Tucán de pico acanalado",
        scientificName: "Ramphastos vitellinus",
        family: "Ramphastidae (Tucanes)",
        statusIUCN: "Vulnerable (VU)"),
    BirdSpecies(
        commonName: "Tucancito esmeralda",
        scientificName: "Aulacorhynchus prasinus",
        family: "Ramphastidae (Tucanes)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Psittacidae
    BirdSpecies(
        commonName: "Guacamaya bandera",
        scientificName: "Ara macao",
        family: "Psittacidae (Loros y Guacamayas)",
        statusIUCN: "Preocupación Menor (LC)"),
    BirdSpecies(
        commonName: "Loro orejiamarillo",
        scientificName: "Ognorhynchus icterotis",
        family: "Psittacidae (Loros y Guacamayas)",
        statusIUCN: "En Peligro Crítico (CR)",
        isEndemic: true),

    // Cotingidae
    BirdSpecies(
        commonName: "Gallito de las rocas",
        scientificName: "Rupicola peruvianus",
        family: "Cotingidae (Gallitos y Cotingas)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Falconidae
    BirdSpecies(
        commonName: "Cernícalo americano",
        scientificName: "Falco sparverius",
        family: "Falconidae (Halcones y Caracaras)",
        statusIUCN: "Preocupación Menor (LC)"),
    BirdSpecies(
        commonName: "Halcón peregrino",
        scientificName: "Falco peregrinus",
        family: "Falconidae (Halcones y Caracaras)",
        statusIUCN: "Preocupación Menor (LC)",
        isMigratorial: true),

    // Columbidae
    BirdSpecies(
        commonName: "Torcaza común",
        scientificName: "Zenaida auriculata",
        family: "Columbidae (Palomas y Tórtolas)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Thraupidae
    BirdSpecies(
        commonName: "Azulejo común",
        scientificName: "Thraupis episcopus",
        family: "Thraupidae (Tangaras y semilleros)",
        statusIUCN: "Preocupación Menor (LC)"),
    BirdSpecies(
        commonName: "Tangara rastrojera",
        scientificName: "Tangara vitriolina",
        family: "Thraupidae (Tangaras y semilleros)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Turdidae
    BirdSpecies(
        commonName: "Mirla común",
        scientificName: "Turdus fuscater",
        family: "Turdidae (Mirlas y tordos)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Ardeidae
    BirdSpecies(
        commonName: "Garza blanca",
        scientificName: "Ardea alba",
        family: "Ardeidae (Garzas)",
        statusIUCN: "Preocupación Menor (LC)"),

    // Cathartidae
    BirdSpecies(
        commonName: "Cóndor andino",
        scientificName: "Vultur gryphus",
        family: "Cathartidae (Buitres y Cóndores)",
        statusIUCN: "Vulnerable (VU)")
  ];

  static List<BirdSpecies> getSpeciesByFamily(String family) {
    return species.where((s) => s.family == family).toList();
  }
}

File: app/src/main/java/com/example/data/SpeciesCatalog.kt
package com.example.data

data class BirdSpecies(
    val commonName: String,
    val scientificName: String,
    val family: String,
    val statusIUCN: String, // LC, NT, VU, EN, CR
    val isEndemic: Boolean = false,
    val isMigratorial: Boolean = false
)

object SpeciesCatalog {
    val families = listOf(
        "Trochilidae (Colibríes)",
        "Ramphastidae (Tucanes)",
        "Psittacidae (Loros y Guacamayas)",
        "Cotingidae (Gallitos y Cotingas)",
        "Falconidae (Halcones y Caracaras)",
        "Columbidae (Palomas y Tórtolas)",
        "Thraupidae (Tangaras y semilleros)",
        "Turdidae (Mirlas y tordos)",
        "Ardeidae (Garzas)",
        "Cathartidae (Buitres y Cóndores)"
    )

    val species = listOf(
        // Trochilidae
        BirdSpecies("Colibrí rutilante", "Colibri coruscans", "Trochilidae (Colibríes)", "Preocupación Menor (LC)"),
        BirdSpecies("Colibrí picoespada", "Ensifera ensifera", "Trochilidae (Colibríes)", "Preocupación Menor (LC)", isEndemic = true),
        BirdSpecies("Colibrí coliancho", "Selasphorus platycercus", "Trochilidae (Colibríes)", "Preocupación Menor (LC)", isMigratorial = true),
        
        // Ramphastidae
        BirdSpecies("Tucán de pico acanalado", "Ramphastos vitellinus", "Ramphastidae (Tucanes)", "Vulnerable (VU)"),
        BirdSpecies("Tucancito esmeralda", "Aulacorhynchus prasinus", "Ramphastidae (Tucanes)", "Preocupación Menor (LC)"),
        
        // Psittacidae
        BirdSpecies("Guacamaya bandera", "Ara macao", "Psittacidae (Loros y Guacamayas)", "Preocupación Menor (LC)"),
        BirdSpecies("Loro orejiamarillo", "Ognorhynchus icterotis", "Psittacidae (Loros y Guacamayas)", "En Peligro Crítico (CR)", isEndemic = true),
        
        // Cotingidae
        BirdSpecies("Gallito de las rocas", "Rupicola peruvianus", "Cotingidae (Gallitos y Cotingas)", "Preocupación Menor (LC)"),
        
        // Falconidae
        BirdSpecies("Cernícalo americano", "Falco sparverius", "Falconidae (Halcones y Caracaras)", "Preocupación Menor (LC)"),
        BirdSpecies("Halcón peregrino", "Falco peregrinus", "Falconidae (Halcones y Caracaras)", "Preocupación Menor (LC)", isMigratorial = true),
        
        // Columbidae
        BirdSpecies("Torcaza común", "Zenaida auriculata", "Columbidae (Palomas y Tórtolas)", "Preocupación Menor (LC)"),
        
        // Thraupidae
        BirdSpecies("Azulejo común", "Thraupis episcopus", "Thraupidae (Tangaras y semilleros)", "Preocupación Menor (LC)"),
        BirdSpecies("Tangara rastrojera", "Tangara vitriolina", "Thraupidae (Tangaras y semilleros)", "Preocupación Menor (LC)"),
        
        // Turdidae
        BirdSpecies("Mirla común", "Turdus fuscater", "Turdidae (Mirlas y tordos)", "Preocupación Menor (LC)"),
        
        // Ardeidae
        BirdSpecies("Garza blanca", "Ardea alba", "Ardeidae (Garzas)", "Preocupación Menor (LC)"),
        
        // Cathartidae
        BirdSpecies("Cóndor andino", "Vultur gryphus", "Cathartidae (Buitres y Cóndores)", "Vulnerable (VU)")
    )

    fun getSpeciesByFamily(family: String): List<BirdSpecies> {
        return species.filter { it.family == family }
    }
}
